includes/admin/class-wc-rest-payments-webhook-controller.php

```diff
@@ -264,6 +264,12 @@ private function process_webhook_payment_intent_succeeded( $event_body ) {
 			);
 		}
 
+		// Save mandate id necessary for some subscriptions renewal.
+		$mandate_id = $event_body['data']['object']['charges']['data'][0]['payment_method_details']['card']['mandate'] ?? null;
```

Comment:
This is quite deep! I opted for the null coalescing operator instead of checking for the existence of each key. Apart from that, not sure if there is a better place to put it. The mandate id is only included in the payment_intent.succeed so we need to rely on the event to get it.

Reply:
Even though it's very deep, I can't really think of a better way to do this.